google/cloud/spanner_dbapi/connection.py

```diff
@@ -252,6 +293,7 @@ def rollback(self):
         elif self._transaction:
             self._transaction.rollback()
             self._release_session()
+            self._statements = []
```

Comment:
When transaction is finished (commited or roll backed) we can drop all the held statements.